docs/API.md

```diff
@@ -28,15 +25,24 @@ class Dirty
 end
 END
 
-source_code = Reek::Source::SourceCode.from(source)
 reporter = Reek::CLI::Report::TextReport.new
-reporter.add_examiner Reek::Core::Examiner.new(source_code)
+examiner = Reek::Examiner.new(source)
+reporter.add_examiner examiner
+
 puts reporter.show
 ```
 
 This will show the list of errors in variable `source`.
 
-`reek` can take `source` as `String`, `File` or `IO`.
+`Reek::Examiner.new` can take `source` as `String`, `File` or `IO`, or an array
+of file names.
+
+```
+# Examine a file object
+reporter.add_examiner Reek::Examiner.new(File.new('dirty.rb'))
+# Examine source from an array of file names
+reporter.add_examiner Reek::Examiner.new(['also_dirty.rb', 'clean.rb')
+```
```

Comment:
Yeah, my first goal was to get a minimal API in place based on the current code, and to only then see what can be improved.

* Regarding an API namespace: I don't think that is needed. We can just expose methods on `Reek`. We can even use those methods ourselves.
* Regarding accessing reporters by a symbol: I think that is a good idea, since it means not having to fix the reports' full class name.
* Rergarding the all-in-one creation of a reporter: I think that would be overkill. Ideally, the publicly exposed API would be something we could use, and do use, inside reek itself.